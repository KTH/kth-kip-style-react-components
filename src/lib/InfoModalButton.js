import React, { Component } from 'react'
import { Button } from 'reactstrap'
import InfoModal from './InfoModal'

class InfoModalButton extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isOpen: false
    }
    this.toggle = this.toggle.bind(this)
  }

  toggle () {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  render () {
    //* *** Example *** *//
    /*
    modalLabels = {
      header: 'Choose image',
      body: `<p>Choose the image that will be displayed on the page Course information.
        You can choose a default image based on the main subject of the course or choose to upload an image on your own choice.
        The image will be displayed with the format 300px * 400px. The file format must be .png or .jpg.</p>`,
      btnClose: 'Close'
    }
    <InfoModalButton
      modalId='infoPic'
      modalLabels={modalLabels}
    />
    */

    const { className, modalId, modalLabels } = this.props // maybe add id
    const { header, body, btnClose } = modalLabels

    return (
      <span className={className}>
        <Button className='btn-info-modal' aria-label='Info'
          onClick={this.toggle}
        />
        <InfoModal isOpen={this.state.isOpen} children={this.props.children} onToggle={this.toggle}
          id={modalId} header={header} htmlBody={body} closeLabel={btnClose} />
      </span>
    )
  }
}

export default InfoModalButton
