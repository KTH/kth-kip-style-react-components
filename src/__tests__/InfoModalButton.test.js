import React from 'react'
import renderer from 'react-test-renderer'
import InfoModalButton from '../lib/InfoModalButton'

const TEST_INFO_FOR_IMAGE = {
  header: 'Choose image',
  body: `<p>Choose the image that will be displayed on the page Course information. 
    You can choose a default image based on the main subject of the course or choose to upload an image on your own choice. 
    The image will be displayed with the format 300px * 400px. The file format must be .png or .jpg.</p>`,
  btnClose: 'Close'
}

describe('InfoModalButton', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<InfoModalButton id='infoPic'
        modalLabels={TEST_INFO_FOR_IMAGE} className='modal-for-info-for-picture' />
      ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
