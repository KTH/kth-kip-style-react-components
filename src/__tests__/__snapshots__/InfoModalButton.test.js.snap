// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModalButton renders properly 1`] = `
<span
  className="modal-for-info-for-picture"
>
  <button
    aria-label="Info"
    className="btn-info-modal btn btn-secondary"
    onClick={[Function]}
    type="button"
  />
</span>
`;
